jvb/src/main/kotlin/org/jitsi/videobridge/stats/StatsManager.kt

```diff
@@ -59,20 +60,24 @@ class StatsManager(statistics: Statistics) {
     val transports: Collection<StatsTransport>
         get() = transportRunnables.map { it.o }
 
+    private val running = AtomicBoolean()
+
     /**
      * Adds a specific <tt>StatsTransport</tt> through which this [StatsManager] is to periodically send statistics.
      *
-     * Warning: `StatsTransport`s added to this `StatsManager` after [start] has been invoked will not be called.
-     *
      * @param transport the [StatsTransport] to add
      * @param period the internal/period in milliseconds at which this [StatsManager] is to repeatedly send statistics
      * to the specified [transport].
      */
     fun addTransport(transport: StatsTransport, period: Long) {
         require(period >= 1) { "period $period" }
 
-        // XXX The field transport is a CopyOnWriteArrayList in order to avoid synchronization here.
-        transportRunnables.add(TransportPeriodicRunnable(transport, period))
+        TransportPeriodicRunnable(transport, period).also {
+            transportRunnables.add(it)
```

Comment:
We do now (see `Main` in the last commit): we start the manager early and pass it to other modules (i.e. callstats and XMPP), which add themselves. I guess we could change this to delay the `start`, but this seems cleaner

Reply:
Ok, passing them in on construction would probably simplify things a bit (no need for this logic and `transportRunnables` could be immutable and not need to be threadsafe).  Is it awkward in Main to pass them in when creating it?

Reply:
There's no good way to do this, because the callstats lib initializes async. The old code creates a `StatsTransport` (that was also serving a bunch of other purposes) early, but this also means it's receiving and ignoring stats until the library loads. If we made the field immutable we would also be unable to remove that transport if callstats fails to load.

Reply:
Ah ha, ok...if this is tied to the CallStats init async callback then I think that makes perfect sense.